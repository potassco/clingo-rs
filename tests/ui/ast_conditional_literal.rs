use clingo::*;

fn main() {
    let loc = ast::Location::default();
    let sym = Symbol::create_id("test", true).unwrap();
    let term1: ast::Term = ast::symbolic_term(&loc, &sym).unwrap().into();
    let term2: ast::Term = ast::symbolic_term(&loc, &sym).unwrap().into();

    let atom1 = ast::symbolic_atom(term1).unwrap();
    let lit = ast::basic_literal_from_symbolic_atom(&loc, ast::Sign::NoSign, atom1).unwrap();

    let atom2 = ast::symbolic_atom(term2).unwrap();
    let lit2: ast::Literal = ast::basic_literal_from_symbolic_atom(&loc, ast::Sign::NoSign, atom2)
        .unwrap()
        .into();

    let condition = vec![lit2];
    let cond = ast::conditional_literal(&loc, lit, &condition).unwrap();

    drop(condition);
    let _end = cond;
}
