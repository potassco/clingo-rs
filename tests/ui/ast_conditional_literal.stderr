error[E0505]: cannot move out of `condition` because it is borrowed
  --> tests/ui/ast_conditional_literal.rs:20:10
   |
17 |     let condition = vec![lit2];
   |         --------- binding `condition` declared here
18 |     let cond = ast::conditional_literal(&loc, lit, &condition).unwrap();
   |                                                    ---------- borrow of `condition` occurs here
19 |
20 |     drop(condition);
   |          ^^^^^^^^^ move out of `condition` occurs here
21 |     let _end = cond;
   |                ---- borrow later used here
